### DISCUSSION TITLE
CrazyFlie 2.x land command not stopping motor

### DISCUSSION THREAD

**Discussion by maichongju** [bitcraze/discussions#74] [Q&A]
As the title suggested, when I call the `land` method from `MotionCommander`. The crazyflie will land but the motor is not stopping. The motor will only stop if I disconnect from the drone.

Here is the sniper of the take-off, land, connect, disconnect method I created. 

```python
    def take_off(self):
        """Crazyflie take off

        Raises:
            DroneException: If connection is not open, or is already flying
        """
        if not self.is_connect:
            raise DroneException("Drone is not connected")
        try:
            self._mc.take_off()
            logger.drone(str(self) + " taking off")
        except Exception as e:
            raise DroneException(str(e))

    def land(self):
        """Crazyflie landing
        """
        if not self.is_connect:
            logger.warning(str(self) + " is not connected")
            return
        elif not self.is_flying:
            logger.warning(str(self) + " is not flying")
            return
        self._mc.land()
        logger.drone(str(self) + " landing")

    def connect(self):
        """Open connection to the Crazyflie

        Raises:
            DroneException: If connection is already open
        """
        self._scf = SyncCrazyflie(self._uri, cf=Crazyflie(rw_cache='./cache'))
        try:
            self._scf.open_link()
            self._scf.wait_for_params()
        except Exception as e:
            raise DroneException(str(e))
        self._logger = Logger(self._scf, 
                            self._log_var,
                            name = self._name,
                            drone_state=self._drone_state, 
                            display_log=False)
        self._logger.start()
        self._mc = MotionCommander(self._scf)
        self._display_connect_msg()

    def disconnect(self):
        """Close connection to the Crazyflie
        """
        if not self.is_connect:
            logger.warning(str(self) + " is not connected")
            return
        
        if self.is_flying:
            self.land()

        self._scf.close_link()
        self._logger.stop()
        self._mc = None
        self._logger = None
        logger.info(str(self) + " disconnected")
```

**Comment by krichardsson** [comment]
Hi!

I'm not sure why the motors do not stop. What versions are you using of the Crazyflie firmware and the python lib?

Would it be possible for you to create a condensed example (with as few lines of code as possible) that shows the problem?

**Reply by maichongju** [reply]
Hi, so here is what I found out. 

Here is the code is used for quick testing. 

```python
URI = uri_helper.uri_from_env(default='radio://0/80/2M/E7E7E7E702')
cflib.crtp.init_drivers()

    cf = Crazyflie(rw_cache='./cache')
    with SyncCrazyflie(URI, cf=cf) as scf:
        with MotionCommander(scf) as motion_commander:
            print("taking off")
            time.sleep(5)
        print("landing")
        time.sleep(5)
        print("disconnected")
```

I am using virtual python environment with python 3.10.4 (anaconda). I install the cflib by using the following command 
`pip install cflib`. And it installed `cflib-0.1.19`

Here is what I found out. If I use the cflib that installed by pip. Then it will land but the motor won't stop. However, if I use the source code from github, then it work fine. I am not sure what is the different between those two. 

To ensure I have not accidentally add or remove any code from the cflib. I reinstall the cflib from pip. But the same behavior still occurring.

Is there any way i can figure out why is not stopping to motor?  

Also another interested behavior is that the program won't quit and will stuck(won't exit) after disconnect.

**Reply by krichardsson** [reply]
Good minimal example, thanks!

I tested on my machine (ubuntu, python 3.10.4) but could not reproduce your findings. Not sure what the problem might be, but possibly related to your environment?

**Reply by maichongju** [reply]
So I switch to another computer. I test with the same program. Now it works. Absolutely no idea why it doesn't work on my laptop.

**Reply by krichardsson** [reply]
Great that it works!
